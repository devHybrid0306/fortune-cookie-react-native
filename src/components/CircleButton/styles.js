import { StyleSheet } from 'react-native';
import { RatioW } from '../../utils';

const styles = StyleSheet.create({
  container: {
    width: 50 * RatioW,
    height: 50 * RatioW,
    borderRadius: 25 * RatioW,
    backgroundColor: '#000000',
    justifyContent: 'center',
    alignItems: 'center',

    shadowColor: '#000',
    shadowOffset: {
      width: 0,
      height: 7,
    },
    shadowOpacity: 0.41,
    shadowRadius: 9.11,

    elevation: 14,
  },
  text: {
    fontSize: 40 * RatioW,
    color: '#ffffff',
  },
});

export default styles;
