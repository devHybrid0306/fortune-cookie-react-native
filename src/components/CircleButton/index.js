import React from 'react';
import { Text, TouchableWithoutFeedback, View } from 'react-native';

import { RatioW } from '../../utils';
import styles from './styles';

const CircleButton = ({
  handleButton,
  size,
  bkgColor = '#000000',
  text,
  textColor = '#ffffff',
  textSize,
}) => {
  return (
    <TouchableWithoutFeedback onPress={handleButton}>
      <View
        style={{
          ...styles.container,
          width: size * RatioW,
          height: size * RatioW,
          borderRadius: (size / 2) * RatioW,
          backgroundColor: bkgColor,
        }}>
        <Text style={{ ...styles.text, color: textColor, fontSize: textSize }}>{text}</Text>
      </View>
    </TouchableWithoutFeedback>
  );
};

export default CircleButton;
