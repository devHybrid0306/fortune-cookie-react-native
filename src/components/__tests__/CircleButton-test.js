import React from 'react';
import renderer from 'react-test-renderer';
import CircleButton from './../CircleButton';

test('renders correctly', () => {
  const tree = renderer.create(<CircleButton />).toJSON();
  expect(tree).toMatchSnapshot();
});
