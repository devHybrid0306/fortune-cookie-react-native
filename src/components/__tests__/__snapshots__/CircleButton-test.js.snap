// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#000000",
      "borderRadius": NaN,
      "elevation": 14,
      "height": NaN,
      "justifyContent": "center",
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": 7,
        "width": 0,
      },
      "shadowOpacity": 0.41,
      "shadowRadius": 9.11,
      "width": NaN,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#ffffff",
        "fontSize": undefined,
      }
    }
  />
</View>
`;
